### DISCUSSION TITLE
LineItems allow a quantity of zero

### DISCUSSION THREAD

**Discussion by RyanofWoods** [solidusio/solidus#5178] [New Features or Ideas]
Adding a LineItem with a quantity of zero to cart doesn't really make sense. There was discussion on removing the normalization of LineItem's quantity to 0, and enforcing the quantity to be 1 and higher. This could free up several tangential parts of the codebase:

https://github.com/solidusio/solidus/pull/4177#discussion_r1012036724

https://github.com/solidusio/solidus/blob/d7b8cf9646b5a3f6bdc9a36662b44b32bd9fbde2/core/app/models/spree/line_item.rb#L25
https://github.com/solidusio/solidus/blob/d7b8cf9646b5a3f6bdc9a36662b44b32bd9fbde2/core/app/models/spree/line_item.rb#L137-L139

**Comment by waiting-for-dev** [comment]
IIRC, the proposal didn't go through because that could break clients relying on that behavior to remove line items. However, I don't see why we can't go through a regular deprecation cycle. We can even put it on hold until the next major if we see it as critical.

**Comment by kennyadsl** [comment]
I'm in favor of this change. I also tried to remove it at some point with https://github.com/solidusio/solidus/pull/4373, but didn't have the proper time to complete the work.

**Reply by RyanofWoods** [reply]
I am working on the two PRs as we previously discussed @kennyadsl. I should have something in the upcoming weeks.
